### DISCUSSION TITLE
[BUG] `segmentation fault` in registration in Apple M1

### DISCUSSION THREAD

**Discussion by Froute99** [dashingsoft/pyarmor#1262] [Q&A]
I've purchased Pyarmor pro license, and tried to register with given registration code. But I got segmentation fault message.
I read previous segmentation fault issue on registration, but it didn't solve my problem.

I'm using macbook-pro, arm64.

```
pyarmor reg -p “CompanyName” pyarmor-regcode-5111.txt
. . .
INFO     process activation file “pyarmor-regcode-5111.txt”
INFO     send request to server
INFO     write registration file “pyarmor-regfile-5111.zip”
INFO     register “pyarmor-regfile-5111.zip”
INFO     update license token
[1]    2707 segmentation fault  pyarmor reg -p “CompanyName” pyarmor-regcode-5111.txt
```

**Comment by jondy** [comment]
First make sure pyarmor is latest version, now it's 8.2.
Then make sure `pytransform3.so` has the right codesign
Third, Pyarmor need libpython3.9.dylib, check it by `otool`
```
$ tool -L ../src-v8/build/darwin.arm64/libs/cp3.9/pytransform3.so
../src-v8/build/darwin.arm64/libs/cp3.9/pytransform3.so:
	pytransform3.so (compatibility version 0.0.0, current version 1.0.0)
	@rpath/lib/libpython3.9.dylib (compatibility version 3.9.0, current version 3.9.0)
...
```

Make sure there is lib pythonX.Y.dylib in the right path, if there is no this file, make a link to `Python`

**Comment by Froute99** [comment]
I checked = pytransform3.so and I've got:
```
	pytransform3.so (compatibility version 0.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.250.1)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1575.17.0)
	@rpath/lib/libpython3.9.dylib (compatibility version 3.9.0, current version 3.9.0)
	/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 946.0.0)
```

Well, actually I've met an error about this before, because I've misset the links to python and pyarmor. So I uninstalled all of them, remove links all unnecessary stuffs.
When I done to set the environments, eventually I've got the seg fault.

I saw a post that saying any dynamic linker environmental variables are purged when launching protected processes.
I tried turning off the SIP on my mac, but it didn't work.

Is there any suggestion or way to solve this seg fault issue?

**Comment by jondy** [comment]
Where's Python 3.9 installed? Using install_name_tool to modify pytransform3.so to link the right Python library, or create a link so that `pytransform3.so` could find Python library.

**Reply by Froute99** [reply]
Well, I'm not very familiar with macOS, and guess I've set something wrong. So I tried to register on virtual machine with linux and a-new-clean-macOS with virtual machine. I've successfully registered in linux, but not the macOS. I still got error
```
INFO     Python 3.9.6
INFO     Pyarmor 8.2.0 (trial), 000000, non-profits
INFO     Platform darwin.aarch64
INFO     register "pyarmor-regfile-5111.zip"
ERROR    "There is no item named 'license.lic' in the archive"
```

**Reply by DarkyAmagai** [reply]
I'm having the same error, idk what is going wrong
